lib/internal/worker.js

```diff
@@ -224,18 +224,19 @@ class Worker extends EventEmitter {
   }
 
   terminate(callback) {
-    if (this[kHandle] === null) return;
-
     debug(`[${threadId}] terminates Worker with ID ${this.threadId}`);
 
     if (typeof callback === 'function') {
       process.emitWarning(
         'Passing a callback to worker.terminate() is deprecated. ' +
         'It returns a Promise instead.',
         'DeprecationWarning', 'DEP0XXX');
+      if (this[kHandle] === null) return;
```

Comment:
Let’s return a promise always. Check out https://www.w3.org/2001/tag/doc/promises-guide#always-return-promises for a reason why.

Reply:
Right, thanks. I'll make the changes